### DISCUSSION TITLE
SSH keys are not added and removed on unlock and close of data base

### DISCUSSION THREAD

**Discussion by frere-jacques** [keepassxreboot/keepassxc#9008] [Q&A] [answered]
I want to use keepassxc as ssh client, since I stored my keypairs there already anyway.
I removed my keys from ~/.ssh and deactivated gnome-keyring which I used before.
I set up everything and get it to run. However it currently works not as expected.
If I go to the entry with my keypair and manually set add to the ssh agent, I can see via `ssh-add -l` that my key is added and I can use the connection. I can also remove them via keepassxc and since I removed them from ~/.ssh I can't connect at all. Thats fine so far.  
However, FAQ states: "It can automatically add SSH keys from your KeePassXC database to a running SSH agent when unlocked and remove them when locked."  
But if I lock my database I can still connect and the key is available. Even if I close keepassxc. Even if I close my session and log in again to the session I have the key in my ssh-agent.

What am I missing?

**Comment by droidmonkey** [comment, marked as answer]
Did you enable that feature in the entry ssh settings to add/remove the key on database unlock and lock?

**Comment by frere-jacques** [comment]
Can you give directions where to find it.

**Reply by droidmonkey** [reply]
https://keepassxc.org/docs/KeePassXC_UserGuide.html#_configuring_an_entry_to_use_ssh_agent

**Reply by frere-jacques** [reply]
Ahhh, my bad. I saw and directly forgot about the per entry config. Cool it works!
Can one make ssh prompt for unlock if database is locked?

**Reply by droidmonkey** [reply]
No, ssh agent doesn't support that signal back to a provider.
